crawler/collector.go

```diff
@@ -121,6 +123,18 @@ func (collector *URLCollector) Collect(rawurl string) []string {
 	return rawurls
 }
 
+func hash(s string) uint64 {
+	h := fnv.New64a()
+	_, err := h.Write([]byte(s))
+
+	if err != nil {
```

Comment:
Do we need to handle errors like this way in entire project? Any other approach of error handling.

Reply:
There are no exceptions in Go, all the errors must be handled ahead of execution. There are panics for runtime errors but thats not relevant here. 
This style of handling errors has mixed opinion in the community though and there is some traction to come up with cleaner syntax in Go2.
see https://github.com/golang/go/issues/21161